velox/functions/prestosql/aggregates/SumAggregate.h

```diff
@@ -43,15 +43,15 @@ class SumAggregate
       folly::Range<const vector_size_t*> indices) override {
     exec::Aggregate::setAllNulls(groups, indices);
     for (auto i : indices) {
-      *exec::Aggregate::value<ResultType>(groups[i]) = 0;
+      *exec::Aggregate::value<TAccumulator>(groups[i]) = 0;
     }
   }
 
   void extractValues(char** groups, int32_t numGroups, VectorPtr* result)
       override {
     BaseAggregate::template doExtractValues<ResultType>(
         groups, numGroups, result, [&](char* group) {
-          return *BaseAggregate::Aggregate::template value<ResultType>(group);
+          return *BaseAggregate::Aggregate::template value<TAccumulator>(group);
```

Comment:
How does this work when TAccumulator <> ResultType? Are you relying on an implicit cast? Would it be better to keep the original code?

Reply:
Yes, the actual intermediate state is stored using TAccumulator type. And it is the type to do reinterpret cast on the storage pointed by value() so it is safer to cast using TAccumulator. The result is returned in ResultType so we need do an implicit conversion between TAccumulator and ResultType.

Reply:
Add an explicit type conversion to make the clear per our offline discussion. Thanks!